### PR TITLE
LPD-20070 | LPD-20078 Fix import of objects with relationships and picklist fields

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related tickets: [LPD-20070](https://liferay.atlassian.net/browse/LPD-20070) & [LPD-20078](https://liferay.atlassian.net/browse/LPD-20078)

### CODE REVIEW

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectStateFlowUtil.java [hunk 1]
```diff
@@ -97,18 +97,18 @@ private static ObjectStateTransition _toObjectStateTransition(
 			ListTypeEntryLocalServiceUtil.fetchListTypeEntry(
 				listTypeDefinitionId, objectStateTransition.getKey());
 
-		if ((objectStateFlowId == 0) && (sourceObjectStateId == 0)) {
+		ObjectState targetObjectState =
```
Comment (G-Sa) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectStateFlowUtil.java:100]: I can see how this improves the code on the scenario where the objectStateFlowId would be 0, but i think this change allows the scenario where the sourceObjectStateId is 0 to still go through. 

That specific scenario happens when we are receiving an object through batch, and so we have the Stateflows but not the Ids set.

Could you try using batch to create an object that has a stateFlow attached and see if the states and the transitions are created as intended?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateLocalServiceImpl.java [hunk 2]
```diff
@@ -69,6 +69,15 @@ public void deleteObjectStateFlowObjectStates(long objectStateFlowId) {
 		objectStatePersistence.removeByObjectStateFlowId(objectStateFlowId);
 	}
 
+	@Override
+	public ObjectState fetchObjectStateFlowObjectState(
+			long listTypeEntryId, long objectStateFlowId)
```
Comment (G-Sa) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateLocalServiceImpl.java:74]: Did this really require a breaking change commit? This doesn't change the major version of the module

**Review by G-Sa — CHANGES_REQUESTED** [review]
Hey @devpedrotavares can you take a look at this? These are my initial thoughts but feel free to reach me if what i said is not clear!
Thanks!

**Comment by G-Sa** [comment]
ci:test:relevant

**Review by G-Sa — APPROVED** [review]

**Comment by devpedrotavares** [comment]
ci:report:2549624462

**Comment by devpedrotavares** [comment]
ci:report:2549644930

**Comment by devpedrotavares** [comment]
ci:forward:force

**Pull request closed without merging**